universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -103,6 +107,29 @@ public String getLocalizedDescription(Locale locale) throws TranslationNotFoundE
     }
   }
 
+  public ImmutableSet<Locale> getSupportedLocales() {
+    ImmutableSet<ImmutableSet<Locale>> questionLocales =
+        getQuestionDefinitions().stream()
+            .map(QuestionDefinition::getSupportedLocales)
+            .collect(toImmutableSet());
+
+    Set<Locale> intersection =
+        Sets.intersection(localizedName().keySet(), localizedDescription().keySet());
+    for (ImmutableSet<Locale> set : questionLocales) {
+      intersection = Sets.intersection(intersection, set);
+    }
+    return ImmutableSet.copyOf(intersection);
+  }
+
+  public ImmutableSet<QuestionDefinition> getQuestionDefinitions() {
```

Comment:
question (non-blocking): Should we make this private until/unless we end up with a need for it to be public?

Reply:
(blocking) this should return a stream since that's how it is used everywhere. It could be renamed `getQuestionDefinitionStream`. And I agree with @natsid that this should be private.

Reply:
Good catch, thank you!